### ISSUE TITLE
Swapchain layer inserting invalid std::unordered_map entries

### ISSUE THREAD

**Issue by dustin-lunarg** [KhronosGroup/Vulkan-LoaderAndValidationLayers#471]
The swapchain layer calls std::unordered_map::operator[] to retrieve records associated with handles without first checking to see if an entry for the handle exists in the map. When no entry is present for the handle, operator[] inserts a new entry with a value-initialized record. This can lead to an invalid pointer dereference with Windows Debug builds, where the memory is initialized with the 0xcdcdcdcd debug pattern. The operator[] calls need to replaced with find() to avoid adding unwanted entries to the map, and the value returned by find() needs to be checked against end() to determine if it was present before it is used.

**Comment by ianelliottus** [comment]
Can you give an example of how I can reproduce this, please?

**Comment by dustin-lunarg** [comment]
You can reproduce it by providing any unrecognized or invalid handle, such as VK_NULL_HANDLE, to the swapchain layer's vkDestroySrufaceKHR function.  The VkLayerTest.EnableWsiBeforeUse test will currently reproduce this error on Windows.

To see what is happening you can call vkDestroySurfaceKHR(instance, 0 , 0) without calling any other WSI functions and set a break point on line [861](https://github.com/KhronosGroup/Vulkan-LoaderAndValidationLayers/blob/master/layers/swapchain.cpp#L861) of the current code.  At break, you will see that my_data->surfaceMap has a size of 0.  After stepping over this line, you will see the size increase to 1.

Basically, you don't want to use operator[] to search for an item in a map, as it has the side effect of inserting new entries for values it does not find.

**Comment by ianelliottus** [comment]
I just created pull request #482 (see: https://github.com/KhronosGroup/Vulkan-LoaderAndValidationLayers/pull/482).  It addresses this and other issues.

**Comment by jon-lunarg** [comment]
This issue has been fixed AFAIK, closing
